client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetMappingsResponseTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.cluster.metadata.MappingMetaData;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import static org.elasticsearch.client.indices.GetMappingsResponse.MAPPINGS;
+import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
+
+public class GetMappingsResponseTests extends ESTestCase {
+
+    public void testFromXContent() throws IOException {
+        xContentTester(
+            this::createParser,
+            GetMappingsResponseTests::createTestInstance,
+            GetMappingsResponseTests::toXContent,
+            GetMappingsResponse::fromXContent)
+            .supportsUnknownFields(true)
+            .randomFieldsExcludeFilter(getRandomFieldsExcludeFilter())
+            .test();
+    }
+
+    private static GetMappingsResponse createTestInstance() {
+        Map<String, MappingMetaData> allMappings = new HashMap<>();
+        allMappings.put("index-" + randomAlphaOfLength(5), randomMappingMetaData());
+        return new GetMappingsResponse(allMappings);
+    }
```

Comment:
I left some of my thinking already in your review of  #37778. Basically, what we really want to test is that we can parse all possible xContent outputs of the server-side response class. It would be great to tie them close together. Copying the randomization methods and converting the client-side instance to a server-side one in order to use its "toXContent" works but has the potential to deviate over time. We should probably discuss this in a larger context than this PR though if we keep continuing with this pattern on server/client side response parsing.

Reply:
👍 I responded there. For now I'm planning on updating this test to generate a random server response, as opposed to a random client one.